k8s/pkg/apis/longhorn/v1beta2/backingimage.go

```diff
@@ -0,0 +1,92 @@
+package v1beta2
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// BackingImageDownloadState is replaced by BackingImageState.
+type BackingImageDownloadState string
+
+type BackingImageState string
+
+const (
+	BackingImageStatePending          = BackingImageState("pending")
+	BackingImageStateStarting         = BackingImageState("starting")
+	BackingImageStateReadyForTransfer = BackingImageState("ready-for-transfer")
+	BackingImageStateReady            = BackingImageState("ready")
+	BackingImageStateInProgress       = BackingImageState("in-progress")
+	BackingImageStateFailed           = BackingImageState("failed")
+	BackingImageStateUnknown          = BackingImageState("unknown")
+)
+
+type BackingImageDiskFileStatus struct {
+	// +optional
+	State BackingImageState `json:"state"`
+	// +optional
+	Progress int `json:"progress"`
+	// +optional
+	Message string `json:"message"`
+	// +optional
+	LastStateTransitionTime string `json:"lastStateTransitionTime"`
+}
+
+// BackingImageSpec defines the desired state of the Longhorn backing image
+type BackingImageSpec struct {
+	// +optional
+	Disks map[string]string `json:"disks"`
+	// +optional
+	Checksum string `json:"checksum"`
+	// +optional
+	SourceType BackingImageDataSourceType `json:"sourceType"`
+	// +optional
+	SourceParameters map[string]string `json:"sourceParameters"`
+	// Deprecated: This kind of info will be included in the related BackingImageDataSource.
+	// +optional
+	ImageURL string `json:"imageURL"`
+}
+
+// BackingImageStatus defines the observed state of the Longhorn backing image status
+type BackingImageStatus struct {
+	// +optional
+	OwnerID string `json:"ownerID"`
+	// +optional
+	UUID string `json:"uuid"`
+	// +optional
+	Size int64 `json:"size"`
+	// +optional
+	Checksum string `json:"checksum"`
+	// +optional
+	DiskFileStatusMap map[string]*BackingImageDiskFileStatus `json:"diskFileStatusMap"`
+	// +optional
+	DiskLastRefAtMap map[string]string `json:"diskLastRefAtMap"`
+	// Deprecated: Replaced by field `State` in `DiskFileStatusMap`.
+	// +optional
+	DiskDownloadStateMap map[string]BackingImageDownloadState `json:"diskDownloadStateMap"`
+	// Deprecated: Replaced by field `Progress` in `DiskFileStatusMap`.
+	// +optional
+	DiskDownloadProgressMap map[string]int `json:"diskDownloadProgressMap"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:shortName=lhbi
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+// +kubebuilder:printcolumn:name="Image",type=string,JSONPath=`.spec.image`,description="The backing image name"
```

Comment:
`.spec.image` is this correct?

Reply:
backingimage's ` .spec.image` also does not exist in v1.1.x and v1.2.x.
Not sure it's a typo or just not implemented.

Reply:
cc @shuo-wu